designer/client/conditions/inline-conditions-absolute-dates.js

```diff
@@ -0,0 +1,204 @@
+import React from 'react'
+import { ConditionValue } from './inline-condition-values'
+import momentTz from 'moment-timezone'
+
+export const absoluteTimeOperators = {
+  is: absoluteTime('=='),
+  'is not': absoluteTime('!='),
+  'is before': absoluteTime('<'),
+  'is after': absoluteTime('>')
+}
+
+export const absoluteDateTimeOperators = {
+  is: absoluteDateTime('=='),
+  'is not': absoluteDateTime('!='),
+  'is before': absoluteDateTime('<'),
+  'is after': absoluteDateTime('>')
+}
+
+export const absoluteDateOperators = {
+  is: absoluteDate('=='),
+  'is not': absoluteDate('!='),
+  'is before': absoluteDate('<'),
+  'is after': absoluteDate('>')
+}
+
+function applyOperator (operator) {
+  return (field, value) => {
+    if (value instanceof ConditionValue) {
+      return `${field.name} ${operator} '${value.toExpression()}'`
+    }
+    throw Error('only Value types are supported')
+  }
+}
+
+function absoluteTime (operator) {
+  return {
+    expression: applyOperator(operator),
+    renderComponent: (value, updateValue) => <AbsoluteTimeValues value={value} updateValue={updateValue} />
+  }
+}
+
+function absoluteDate (operator) {
+  return {
+    expression: applyOperator(operator),
+    renderComponent: (value, updateValue) => <AbsoluteDateValues value={value} updateValue={updateValue} />
+  }
+}
+
+function absoluteDateTime (operator) {
+  return {
+    expression: applyOperator(operator),
+    renderComponent: (value, updateValue) => <AbsoluteDateTimeValues value={value} updateValue={updateValue} />
+  }
+}
+
+class AbsoluteTimeValues extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      hours: props.value?.value.split(':')[0],
+      minutes: props.value?.value.split(':')[1]
+    }
+  }
+
+  componentDidUpdate (prevProps, prevState, snapshot) {
+    const { hours, minutes } = this.state
+    const everythingComplete = hours && minutes
+    const hasChanged = hours !== prevState.hours || minutes !== prevState.minutes
+    if (everythingComplete && hasChanged) {
+      this.props.updateValue(new ConditionValue(`${hours}:${minutes}`))
```

Comment:
could do this.props.updateValue = this.updateParentValue in the constructor so it's clearer as to what the code is doing but fine as is

Reply:
Will do this in a separate PR as it's common across a number of react components